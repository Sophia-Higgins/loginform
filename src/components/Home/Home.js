import React, { useEffect } from "react";
import { useHistory, useLocation } from "react-router-dom";
import { API_BASE_URL, ACCESS_TOKEN_NAME } from "../../constants/apiConstants";
import axios from "axios";

const Home = () => {
  let history = useHistory();
  let location = useLocation();

  useEffect(() => {
    axios
      .get(API_BASE_URL + "/verifyToken", {
        headers: {
          Authorization: "Bearer " + localStorage.getItem(ACCESS_TOKEN_NAME),
        },
      })
      .then((response) => {
        if (response.status !== 200) {
          redirectToLogin();
        }
      })
      .catch(() => {
        redirectToLogin();
      });
  });

  function redirectToLogin() {
    history.push("/login");
  }

  return (
    <div className="body">
      <br />
      <h1 className="display-1 text-center text-primary">
        Welcome {location.state.username}! <br /> Everything is fine.
      </h1>
    </div>
  );
};

export default Home;
