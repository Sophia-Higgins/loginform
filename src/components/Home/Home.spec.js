import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "./Home";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Home />);
});

describe("<Header />", () => {
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
