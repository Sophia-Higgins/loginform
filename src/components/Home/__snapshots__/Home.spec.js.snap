// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
<div
  className="body"
>
  <br />
  <h1
    className="display-1 text-center text-primary"
  >
    Welcome 
    test
    ! 
    <br />
     Everything is fine.
  </h1>
</div>
`;
